src/core/components/NavMenu/NavMenu.tsx

```diff
@@ -1,86 +1,48 @@
-import { Link, useLocation } from 'react-router-dom';
-import { useAppDispatch } from '../../../config/redux/hooks';
-import { clearUserData } from '../../../user/redux/userSlice';
-import { routes } from '../../routing/routes';
-import { CustomButton } from '../../components/buttons/Button/Button';
-import HomeIcon from '@mui/icons-material/Home';
-import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
-import LogoutIcon from '@mui/icons-material/Logout';
-import { useState, MouseEvent } from 'react';
-import { PendingActions } from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import { NAV_MENU, NavMenuItemType } from '../../../config/menuConfig';
+import { useTranslation } from 'react-i18next';
+import { HomeIcon } from '../../../assets/icons';
+import { IconButtonWithLabel } from '../buttons/IconButton';
+import { COLOR_VARIANTS } from '../../types/ColorVariants';
 import './NavMenu.css';
 
-export const NavMenu = (): JSX.Element => {
-  const { pathname } = useLocation();
-  const dispatch = useAppDispatch();
+interface NavMenuProps {
+  activeRoute: string;
+  icons: unknown;
+  onClick?: (url: string) => void;
+}
 
-  const getActiveRouteClass = (route: string) => {
-    const activeRouteClassName = 'font-bold';
+export const NavMenu = ({ activeRoute, icons, onClick }: NavMenuProps): JSX.Element => {
+  const { t } = useTranslation();
 
-    if (route === pathname) return activeRouteClassName;
-
-    if (route === '/') return '';
-
-    const isActiveRoute = new RegExp(route, 'ui').test(pathname);
-
-    return isActiveRoute ? activeRouteClassName : '';
+  const getIcon = (iconName: unknown): JSX.Element => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    //@ts-ignore icons type
+    return !!icons[iconName as keyof typeof icons] ? icons[iconName as keyof typeof icons]() : <HomeIcon />;
```

Comment:
If we define icons as` { [name: string]: () => JSX.Element }`  in the interface then the return statement can be simplified to 
`return !!icons[iconName] ? icons[iconName]() : <HomeIcon />;`

Reply:
Changed.